src/tools/interop/idt/capture/base.py

```diff
@@ -30,19 +30,32 @@ def __init__(self, artifact_dir: str) -> None:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    async def connect(self) -> None:
+        """
+        Establish connections to log targets for this platform
+        """
+        raise NotImplementedError
+
     @abstractmethod
     async def start_streaming(self) -> None:
         """
         Begin streaming logs
-        Start should be able to be called repeatedly without restarting streaming
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    async def run_observers(self) -> None:
+        """
+        Observe log procs and restart as needed
+        Must be async aware and not interact with stdin
         """
         raise NotImplementedError
 
     @abstractmethod
     async def stop_streaming(self) -> None:
         """
-        Stop streaming logs
-        Stop should not cause an error even if the stream is not running
+        Stop the capture and pull any artifacts from remote devices
```

Comment:
Where are these pulled to? Is there a getter function (ie pulled to ram?) or are they dumped to a specific location on disk?

Reply:
They are meant to be written to the artifact directory passed to the ecosystem / platform on instantiation. I can make sure this comment covers that.

Reply:
Added this to the comment!